mathics/builtin/inference.py

```diff
@@ -273,9 +272,6 @@ def get_assumption_rules_dispatch(evaluation):
     if val_consistent_assumptions == SymbolFalse:
         evaluation.message("Inconsistent assumptions")
 
-    if assumptions_list is None:
```

Comment:
Not sure about this one.

Reply:
`pred` isn't defined in this function, so if it be `None`, an undeclared variable exception would be raised.

Reply:
OK, if the CI does not complains, then go ahead. At some point, I have to go through this code again anyway